src/core/Jalasoft.TeamUp.Resumes.Core/ResumesService.cs

```diff
@@ -17,17 +18,48 @@ public ResumesService(IRepository<Resume> resumesRepository)
 
         public Resume GetResume(Guid id)
         {
-            return this.resumesRepository.GetById(id);
+            try
+            {
+                var resume = this.resumesRepository.GetById(id);
+                if (resume == null)
+                {
+                    throw new ResumeException(ErrorsTypes.NotFoundError);
+                }
+
+                return resume;
+            }
+            catch (ResumeException ex)
+            {
+                throw ex;
+            }
+            catch (Exception)
+            {
+                throw new ResumeException(ErrorsTypes.ServerError);
+            }
         }
 
         public Resume[] GetResumes()
         {
-            return this.resumesRepository.GetAll().ToArray();
+            try
+            {
+                return this.resumesRepository.GetAll().ToArray();
+            }
+            catch (Exception)
+            {
+                throw new ResumeException(ErrorsTypes.ServerError);
+            }
         }
 
         public Resume PostResumes(Resume resume)
         {
-            return this.resumesRepository.Add(resume);
+            try
+            {
+                return this.resumesRepository.Add(resume);
+            }
+            catch (Exception)
```

Comment:
please catch everything in the functions

Reply:
handle unhandled exceptions at functions level, here we are losing the stack trace